src/pages/blog/tag/[tag]/[...page].astro

```diff
@@ -0,0 +1,99 @@
+---
+import BaseLayout from "@layouts/BaseLayout.astro";
+import HorizontalCard from "@components/HorizontalCard.astro";
+import { getCollection } from "astro:content";
+
+export async function getStaticPaths({ paginate }) {
+    const all_posts = await getCollection("blog");
+    const all_tags = all_posts.flatMap((post) => {
+        return post.data.tags || [];
+    });
+
+    return all_tags.flatMap((tag) => {
+        const filtred_posts = all_posts.filter((post) => {
+            return post.data.tags?.includes(tag);
+        });
+        return paginate(filtred_posts, {
+            params: { tag },
+            pageSize: 10,
+        });
+    });
+}
+
+const { page } = Astro.props;
+const params = Astro.params;
+---
+
+<BaseLayout title="Blog" sideBarActiveItemID="blog">
```

Comment:
Here we can add a dynamic Title according to the tag selected. Something like `Blog - [tagname]`